chromium_src/chrome/browser/notifications/notification_handler.h

```diff
@@ -12,15 +12,25 @@
 
 class NotificationHandler : public NotificationHandler_ChromiumImpl {
  public:
+  // If you add to the enum make sure the DCHECK in the constructor below is
+  // updated.
   enum class Type {
     WEB_PERSISTENT = 0,
     WEB_NON_PERSISTENT = 1,
     EXTENSION = 2,
     SEND_TAB_TO_SELF = 3,
     TRANSIENT = 4,
-    BRAVE_ADS = 5,
+    PERMISSION_REQUEST = 5,
+    SHARING = 6,
+    BRAVE_ADS = 7,
     MAX = BRAVE_ADS,
   };
+
+  // Make sure we know if the original enum gets changed.
+  NotificationHandler() {
+    DCHECK(static_cast<int>(NotificationHandler_ChromiumImpl::Type::MAX) + 1 ==
+           static_cast<int>(Type::MAX));
```

Comment:
This is a little bit awkward - not sure I see the relationship here between the parent (`NotificationHandler_ChromiumImpl`) classes version of the Type enum and this above implementation. Basically, parent one has MAX as 6? and this will fail if they update... What would the update to the DCHECK look like if you did add an item?

Reply:
yes, this is a bit flimsy. Maybe a better check would be
`DCHECK(static_cast<int>(NotificationHandler_ChromiumImpl::Type::MAX) + 1 ==
           static_cast<int>(Type::BRAVE_ADS));`

Reply:
Addressed by 7d6f8852da936e047acad4a30439ab66be5d5b1d